ionc/ion_reader_binary.c

```diff
@@ -1006,7 +1006,17 @@ iERR _ion_reader_binary_read_int64(ION_READER *preader, int64_t *p_value)
     IONCHECK(ion_binary_read_uint_64(preader->istream, len, &unsignedInt64));
 
     is_negative = (tid == TID_NEG_INT)? TRUE: FALSE;
-    IONCHECK(cast_to_int64(unsignedInt64, is_negative, p_value));
+    iERR result = cast_to_int64(unsignedInt64, is_negative, p_value);
+
+    // if an error is thrown reset reader stream's current pointer to the beginning of current value again
+    if(result) {
+        i = len;
+        while(i--) {
+            preader->istream->_curr--;
```

Comment:
I don't think you need a loop, can't you just `preader->istream->_curr -= len;`?